props/saber_BC_buttons.h

```diff
@@ -1587,39 +1579,70 @@ class SaberBCButtons : public PROP_INHERIT_PREFIX PropBase {
   }
 #endif
 
-  void DetectMenuTurn() {
-    float a = fusor.angle2() - current_twist_angle_;
-    if (isPointingUp()) return;
-    // Keep the rotational angle within range of 
-    // -180 to 180 degrees in terms of radians.
-    if (a > M_PI) a-= M_PI * 2;
-    if (a < -M_PI) a+= M_PI * 2;
-
-    if (a < -M_PI / 3) {
-      CheckSavedTwist(EVENT_TWIST_LEFT);
-      current_twist_angle_ = fusor.angle2();
-    } else if (a > M_PI / 3) {
-      CheckSavedTwist(EVENT_TWIST_RIGHT);
-      current_twist_angle_ = fusor.angle2();
+  // Track the last twist type and time
+  StrokeType saved_twist = UNKNOWN_GESTURE;
+  uint32_t saved_twist_start_millis = 0;
+
+  // Adds a timer to TWIST_LEFT/RIGHT to avoid interference
+  // with normal, 2 direction twists
+  void DetectTwist() override {
```

Comment:
This function is ~75% the same as PropBase::DetectTwist, right?
How about we just break PropBase::DetectTwist into three functions, where the third one would just look like this:

void DetectTwist() {
   bool process = DetectTwistStrokes();
   if (process) {
       ProcessTwistEvents();
   }
}

Then your DetectTwist() can re-use ProcessTwistEvents() and DetectTwistStrokes().
This reduces the amount of duplication, and it also makes it really obvious what the new part is in your function.

Reply:
Ok refactored. How about now?